_posts/2019-01-10-disobey-puzzle-with-powershell.md

```diff
@@ -0,0 +1,283 @@
+---
+layout: post
+title: Solving Disobey 2019 puzzle with PowerShell!
+author: Rinorragi
+excerpt: I was able to claim my hacker badge for Disobey 2019 mainly using PowerShell.
+tags:
+- disobey
+- hacking
+- infosec
+- software security
+- devsecops
+- PowerShell
+---
+
+## Disobey 
+
+[Disobey](https://disobey.fi/) is a Nordic security event. It is all about gathering people from different organizations around security topics. The atmosphere in the disobey is hacker friendly. Many are interested on the techniques to attack systems. Most of the participants are white hat hackers that try to find vulnerabilities that could be then patched or limited with other measures. We have participated Disobey in [2017](https://dev.solita.fi/2017/01/19/Disobey.html) and [2018](https://dev.solita.fi/2018/01/26/Disobey.html).
+
+## Puzzle
+
+As the event is hacker friendly it possible to get a small discount by hacking. Each year there has been a puzzle that you could solve with hacking. This years [puzzle](http://puzzle.disobey.fi/) was frustrating, fun and teached me a bunch about hacking. My coworker was able to solve it by creating some utility software with python: [https://teamrot.fi/2018/09/17/solving-the-disobey-2018-puzzle/](https://teamrot.fi/2018/09/17/solving-the-disobey-2018-puzzle/). I instead took a painful route by trying the same thing with PowerShell. It is not that PowerShell would not be good but I was kind of alone with the whole "hacking with PowerShell" mentality. Without the encouragment from [Team Rot](https://teamrot.fi) members I would not have finished it. 
+
+This blog post is about sharing my experiences on "hacking with PowerShell". Spoiler alert.
+
+## Starting with quick recon
+
+Running nmap with quick scan reveals that there are few ports open in the server:
+* 80/tcp
+* 443/tcp
+* 8021/tcp 
+
+After spending many hours with the 8021 port I gave up and went back to the orginal web page. 
+
+## Dirbusting 
+
+Using common folder name list from [https://github.com/danielmiessler/SecLists/blob/master/Discovery/Web-Content/common.txt](https://github.com/danielmiessler/SecLists/blob/master/Discovery/Web-Content/common.txt) we start trying to find interesting addresses. I recall that I used OWASP ZAP for this and found .bash_history that lead to a secret url [http://puzzle.disobey.fi/sicret_admin_address/](http://puzzle.disobey.fi/sicret_admin_address/). There is a juicy looking file called mysql_backup.db which after a few hours turned out to be a dead end.
+
+## Page source and lorem.html
+
+Back to the front page of the puzzle. Viewing source for puzzle.disobey.fi shows that there is a somewhat hidden link in the page for the /lorem.html. When the lorem.html is opened there is just page full of lorem ipsum. 
+
+Storing the lorem ipsum into original_lorem.txt we can use the content for dirbusting. Most of the times the call seems to response 404 so I ended up with a script that tries to find out if there non-404 answers.
```

Comment:
"The call seems to response 404" ei oikein toimi. Kelpaisiko "Most of the time, the response seems to be 404, so I ended up writing a script that tries to find non-404 answers."